README.md

```diff
@@ -363,12 +364,43 @@ match (res) {
 }
 ```
 
+Here, `Option.Some` might describe a "constructor" function that produces an object discriminated by either a
+well-known symbol field or merely by its \[\[Prototype\]\], such that `Option.Some(0) instanceof Option.Some` is
+`true`. ADT enum members could also describe more complex shapes through the use of binding patterns, such as:
+
+```js
+enum Geometry {
+  Point({ x, y }),
+  Line(p1, p2),
+}
+
+const p1 = Geometry.Point({ x: 0, y: 1 });
+p1[0].x; // 0
+p1[0].y; // 1
+
+const p2 = Geometry.Point({ x: 2, y: 3 });
+const l = Geometry.Line(p1, p2);
+l[0] === p1; // true
+
+const printGeom = geom => match (geom) {
+  Geometry.Point({ let x, let y }): console.log(`Point({ x: ${x}, y: ${y} })`);
+  Geometry.Line(let p1, let p2): console.log(`Line(${printGeom(p1)}, ${printGeom(p2)})`);
+};
+
+printGeom(p1); // Point({ x: 0, y: 1 })
+printGeom(l); // Line(Point({ x: 0, y: 1 }), Point({ x: 2, y: 3 }))
+```
+
+ADT enum members may also need a mechanism to implement prototypal or static methods on the `enum`, which is one
+reason why we prefer `@@iterator` to describe the domain of an `enum` vs. something like `Object.entries()`.
```

Comment:
nits: Notations such as `@@iterator` are no longer used in the spec so it may be better to change it to `%Symbol.iterator%`.
https://github.com/tc39/ecma262/pull/1314